<launch>
  <!-- Arguments -->
  <arg name="nodelet" default="true"/>

  <arg name="left_side_usensor_frame_id"    default="left_side_usensor_link"/>
  <arg name="left_front_usensor_frame_id"   default="left_front_usensor_link"/>
  <arg name="right_front_usensor_frame_id"  default="right_front_usensor_link"/>
  <arg name="right_side_usensor_frame_id"   default="right_side_usensor_link"/>
  <arg name="usensor_hight_noise_threshold" default="4000"/>

  <!-- nodelet -->
  <group if="$(arg nodelet)">
    <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" output="screen"/>
    
    <node pkg="nodelet" type="nodelet" name="analog_distance_sensor_to_laserscan_nodelet"
      args="load analog_distance_sensor_to_laserscan/analog_distance_sensor_to_laserscan nodelet_manager"
      output="screen">
      <param name="left_side_usensor_frame_id"      value="$(arg left_side_usensor_frame_id)"/>
      <param name="left_front_usensor_frame_id"     value="$(arg left_front_usensor_frame_id)"/>
      <param name="right_front_usensor_frame_id"    value="$(arg right_front_usensor_frame_id)"/>
      <param name="right_side_usensor_frame_id"     value="$(arg right_side_usensor_frame_id)"/>
      <param name="usensor_hight_noise_threshold"   value="$(arg usensor_hight_noise_threshold)"/>
    </node>
  </group>
  
  <!-- node -->
  <group unless="$(arg nodelet)">
    <node pkg="nodelet" type="nodelet" name="analog_distance_sensor_to_laserscan_node"
      args="standalone analog_distance_sensor_to_laserscan/analog_distance_sensor_to_laserscan"
      output="screen">
      <param name="left_side_usensor_frame_id"      value="$(arg left_side_usensor_frame_id)"/>
      <param name="left_front_usensor_frame_id"     value="$(arg left_front_usensor_frame_id)"/>
      <param name="right_front_usensor_frame_id"    value="$(arg right_front_usensor_frame_id)"/>
      <param name="right_side_usensor_frame_id"     value="$(arg right_side_usensor_frame_id)"/>
      <param name="usensor_hight_noise_threshold"   value="$(arg usensor_hight_noise_threshold)"/>
    </node>
  </group>
</launch>